src/StaticAnalysisResultsInterchangeFormat/GrammarFiles/SarifFormat.g4

```diff
@@ -619,11 +573,243 @@ region :
     */
     length?;
 
+/**
+    @className {LogicalLocationComponent}
+    @summary {One level (e.g. namespace, function, etc.) of a logical location tree.}
+*/
+logicalLocationComponent:
+    /**
+        @name {Name}
+        @summary {
+        Name of the item specified by this location component.
+        }
+    */
+    name
+
+    /**
+        @name {LocationKind}
+        @summary {
+        The type of item this location refers to.
+        }
+        @remarks {
+        Examples include namespace, class, or function.
+        }
+    */
+    locationKind;
+
+/**
+    @className {AnnotatedCodeLocation}
+    @summary {
+    A code annotation that consists of single physical location and associated message, used to express
+	stacks, document execution flow through a method, etc.
+    }
+*/
+annotatedCodeLocation:
+	/**
+		@name {PhysicalLocations} @serializedName{physicalLocations}
+		@summary {
+		A set of places to which this annotation refers. These locations are of equal
+		priority; for example, one entry may be the source code location while another is the
+		assembly or IL location.
+		}
+	*/
+	physicalLocations
+
+	/**
+		@name {Message} @serializedName {message}
+		@summary {
+		A message associated with this annotation, if applicable.
+		}
+	*/
+	message?;
+
+/** @className {AnnotatedCodeLocations}
+    @summary {
+    A set of one or more annotated code locations associated with a given result.
+    }
+*/
+annotatedCodeLocations: annotatedCodeLocation*;
+
+/*********************************************************
+Execution flow infrastructure; execution flows are the same lists of physical locations except that
+each physical location in the list can have a message stapled on.
```

Comment:
I'm not fond of the tone of "stapled on", and in any case I'm not sure how valuable this whole sentence (which I know is pre-existing) is.